shuttle_test.go

```diff
@@ -276,20 +334,19 @@ func BenchmarkPipeline(b *testing.B) {
 	config.LogsURL = ts.URL
 	config.InputFormat = InputFormatRaw
 
-	b.ResetTimer()
+	shut := NewShuttle(config)
+	input := NewLoopingBuffer(longerTestData)
+	shut.LoadReader(input)
+	shut.Launch()
+	var tb int
 	for i := 0; i < b.N; i++ {
-		b.StopTimer()
-		shut := NewShuttle(config)
-		input := NewLongerTestInput()
-		b.SetBytes(int64(input.Len()))
-
-		shut.LoadReader(input)
-
-		b.StartTimer()
-		shut.Launch()
-		shut.WaitForReadersToFinish()
-		shut.Land()
+		// This sleep is here to allow the reading goroutines to make
```

Comment:
Could have a testing hook function that is normally a noop but can be set to something when testing to indicate something happened. Similar to [this](https://github.com/golang/go/blob/0436a89a2c5afad41356dc1dff7c745cd30636a7/src/net/http/transport.go#L376) in net/http.